notifiers/toaster.js

```diff
@@ -51,36 +67,45 @@ WindowsToaster.prototype.notify = function(options, callback) {
     );
   }
 
-  var actionJackedCallback = utils.actionJackerDecorator(
-    this,
-    options,
-    function cb(err, data) {
-      /* Possible exit statuses from SnoreToast, we only want to include err if it's -1 code
-      Exit Status     :  Exit Code
-      Failed          : -1
-
-      Success         :  0
-      Hidden          :  1
-      Dismissed       :  2
-      TimedOut        :  3
-      ButtonPressed   :  4
-      TextEntered     :  5
-      */
-      if (err && err.code !== -1) {
-        return callback(null, data);
-      }
-      callback(err, data);
-    },
-    function mapper(data) {
-      if (hasText(data, successMessage)) {
-        return 'click';
-      }
-      if (hasText(data, timeoutMessage)) {
-        return 'timeout';
-      }
-      return false;
+  var snoreToastResultParser = (err, callback) => {
+    /* Possible exit statuses from SnoreToast, we only want to include err if it's -1 code
+    Exit Status     :  Exit Code
+    Failed          : -1
+
+    Success         :  0
+    Hidden          :  1
+    Dismissed       :  2
+    TimedOut        :  3
+    ButtonPressed   :  4
+    TextEntered     :  5
+    */
+    const result = parseResult(
+      resultBuffer && resultBuffer.toString('utf16le')
+    );
+
+    // parse action
+    if (result.action === 'buttonClicked' && result.button) {
+      result.activationType = result.button;
+    } else if (result.action) {
+      result.activationType = result.action;
     }
-  );
+
+    if (err && err.code === -1) {
+      callback(err, result);
+    }
+    callback(null, result);
+  };
+
+  var actionJackedCallback = err =>
+    snoreToastResultParser(
+      err,
+      utils.actionJackerDecorator(
+        this,
+        options,
+        callback,
+        data => data || false
```

Comment:
We probably should fix the mapper instead of having it be an identity function. You probably didn't notice the mapper actually doing anything since it wasn't working without the pipe you added to get the data back. The mapper function is to make it so we can do the event listeners described in the README, but I can do that in a different PR after this one gets merged :)

Reply:
The identity function was meant to allow custom actions. You can see in my toaster-with-actions.js example where I used both "built-in" actions ("timeout", "activate" & "dismissed") and custom actions defined by the action option given to the the notify function ("ok" & "cancel"). Those custom actions can be anything. (Name of the action is the lower-case of the button name.)

Reply:
Ah, that makes sense. I didn't realize that the buttons were the source of those custom actions in the example. 

I'll leave this as is then :)